libc++ test runner fails on Python 3.13: `ModuleNotFoundError: No module named 'pipes'`

```pytb
$ cmake ../runtimes/ -G Ninja -DLLVM_ENABLE_RUNTIMES='libcxx;libcxxabi;libunwind'
[…]
$ ninja
[…]
$ ninja check-cxx
[0/1] Running libcxx tests
llvm-lit: /home/mgorny/git/llvm-project/llvm/utils/lit/lit/TestingConfig.py:152: fatal: unable to parse config file '/home/mgorny/git/llvm-project/build/libcxx/test/lit.site.cfg', traceback: Traceback (most recent call last):
  File "/home/mgorny/git/llvm-project/llvm/utils/lit/lit/TestingConfig.py", line 140, in load_from_path
    exec(compile(data, path, "exec"), cfg_globals, None)
    ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/mgorny/git/llvm-project/build/libcxx/test/lit.site.cfg", line 21, in <module>
    import libcxx.test.params, libcxx.test.config
  File "/home/mgorny/git/llvm-project/libcxx/utils/libcxx/test/params.py", line 13, in <module>
    from libcxx.test.dsl import *
  File "/home/mgorny/git/llvm-project/libcxx/utils/libcxx/test/dsl.py", line 11, in <module>
    import pipes
ModuleNotFoundError: No module named 'pipes'

FAILED: libcxx/test/CMakeFiles/check-cxx /home/mgorny/git/llvm-project/build/libcxx/test/CMakeFiles/check-cxx 
cd /home/mgorny/git/llvm-project/build/libcxx/test && /usr/bin/python3.13 /home/mgorny/git/llvm-project/build/bin/llvm-lit -sv --show-xfail --show-unsupported /home/mgorny/git/llvm-project/build/libcxx/test
ninja: build stopped: subcommand failed.
```

I'm going to try making a pull request to fix it.

--- a/libcxx/test/libcxx/lit.local.cfg
+++ b/libcxx/test/libcxx/lit.local.cfg
@@ -1,4 +1,5 @@
 # The tests in this directory need to run Python
-import pipes, sys
+import shlex
+import sys
 
-config.substitutions.append(("%{python}", pipes.quote(sys.executable)))
+config.substitutions.append(("%{python}", shlex.quote(sys.executable)))

--- a/libcxx/utils/libcxx/test/dsl.py
+++ b/libcxx/utils/libcxx/test/dsl.py
@@ -8,8 +8,8 @@
 
 import os
 import pickle
-import pipes
 import platform
+import shlex
 import shutil
 import tempfile
 
@@ -290,7 +290,7 @@ def hasAnyLocale(config, locales):
       }
     #endif
   """
-    return programSucceeds(config, program, args=[pipes.quote(l) for l in locales])
+    return programSucceeds(config, program, args=[shlex.quote(l) for l in locales])
 
 
 @_memoizeExpensiveOperation(lambda c, flags="": (c.substitutions, c.environment, flags))

